tests/countries/test_algeria.py

```diff
@@ -12,161 +12,341 @@
+            "2020-05-24",
+            "2020-05-25",
+            "2021-05-13",
+            "2021-05-14",
+            "2022-05-02",
+            "2022-05-03",
             "2023-04-21",
             "2023-04-22",
+            "2024-04-10",
+            "2024-04-11",
+            "2024-04-12",
+            "2025-03-30",
+            "2025-03-31",
+            "2025-04-01",
         )
-
-        # Eid al-Adha - Scarfice Festive
-        self.assertNoHoliday(
-            "2023-06-27",
-            "2023-07-02",
-            "2024-07-15",
-            "2024-07-19",
+        self.assertHolidayNameCount(
+            name,
+            2,
+            self.no_estimated_holidays,
+            range(1964, 1968),
+            range(1969, 2000),
+            range(2001, 2024),
+        )
+        self.assertHolidayNameCount(
+            name, 3, self.no_estimated_holidays, range(2024, 2033), range(2034, 2050)
         )
-        self.assertHoliday(
+        self.assertHolidayNameCount(name, 4, self.no_estimated_holidays, 1968, 2000)
+        self.assertHolidayNameCount(name, 6, self.no_estimated_holidays, 2033)
+
+    def test_eid_al_adha(self):
+        name = "عيد الأضحى"
+        self.assertHolidayName(
+            name,
+            self.no_estimated_holidays,
+            "2020-07-31",
+            "2020-08-01",
+            "2021-07-20",
+            "2021-07-21",
+            "2022-07-09",
+            "2022-07-10",
             "2023-06-28",
             "2023-06-29",
             "2023-06-30",
             "2024-06-16",
             "2024-06-17",
             "2024-06-18",
+            "2025-06-06",
+            "2025-06-07",
+            "2025-06-08",
+        )
+        self.assertHolidayNameCount(name, 1, self.no_estimated_holidays, range(1964, 1969))
+        self.assertHolidayNameCount(
+            name,
+            2,
+            self.no_estimated_holidays,
+            range(1969, 1974),
+            range(1975, 2006),
+            range(2009, 2023),
+        )
+        self.assertHolidayNameCount(
+            name,
+            3,
+            self.no_estimated_holidays,
+            range(2006, 2008),
+            range(2023, 2039),
+            range(2040, 2050),
         )
+        self.assertHolidayNameCount(name, 4, self.no_estimated_holidays, 1974)
+        self.assertHolidayNameCount(name, 6, self.no_estimated_holidays, 2039)
 
-        # Islamic New Year
-        self.assertHoliday(
-            "2008-01-10",
-            "2008-12-29",
-            "2020-08-20",
+    def test_easter_monday(self):
+        name = "إثنين الفصح"
+        self.assertHolidayName(
+            name,
+            self.christian_holidays,
+            "2020-04-13",
+            "2021-04-05",
+            "2022-04-18",
+            "2023-04-10",
+            "2024-04-01",
+            "2025-04-21",
         )
+        self.assertHolidayName(name, self.christian_holidays, range(1964, 2050))
+        self.assertNoHolidayName(name)
```

Comment:
In most countries, this check (`self.assertNoHolidayName(name)`) is placed at the beginning of the method, immediately after `name = ...` Maybe we should keep this order?
That's my only suggestion, everything else LGTM. :)